libs/hwes/eventstoredb/replay.go

```diff
@@ -0,0 +1,111 @@
+package eventstoredb
+
+import (
+	"context"
+	"github.com/EventStore/EventStore-Client-Go/v4/esdb"
+	zlog "github.com/rs/zerolog"
+	"hwes"
+	"telemetry"
+)
+
+// Replay provides the ability to re-run existing events from an event stream
+// from the first to the latest event. The latest event will be computed via a read
+// against the event stream during a call the .Run().
+// Users of Replay can use the provided hook to run actions before and after the execution of Replay().
+type Replay struct {
```

Comment:
what is a Replay? Why is this a struct? We can replace a lot of this code by just exporting the Run function (under a better name)